src/earn/prepareTransactions.ts

```diff
@@ -112,106 +114,130 @@ export async function prepareWithdrawAndClaimTransactions({
+      ]),
+      isGasSubsidized: isGasSubsidizedForNetwork(networkId),
+      origin: 'earn-withdraw',
+    }),
+  }
 }
 
 export async function prepareWithdrawTransactions({
   amount,
-  token,
   walletAddress,
   feeCurrencies,
   pool,
   hooksApiUrl,
-  shortcutId,
   useMax,
 }: {
   amount: string
-  token: TokenBalance
   walletAddress: Address
   feeCurrencies: TokenBalance[]
   pool: EarnPosition
   hooksApiUrl: string
-  shortcutId: Exclude<EarnActiveMode, 'exit'>
   useMax: boolean
 }) {
-  const { dataProps } = pool
+  const { appId, balance, dataProps, networkId, shortcutTriggerArgs } = pool
   const { transactions }: { transactions: RawShortcutTransaction[] } = await triggerShortcutRequest(
     hooksApiUrl,
     {
       address: walletAddress,
-      appId: pool.appId,
-      networkId: pool.networkId,
-      shortcutId,
+      appId,
+      networkId,
+      shortcutId: 'withdraw',
       tokens: [
         {
           tokenId: dataProps.withdrawTokenId,
-          amount,
+          amount: useMax ? balance : amount,
           useMax,
         },
       ],
-      ...pool.shortcutTriggerArgs?.[shortcutId],
+      ...shortcutTriggerArgs?.withdraw,
     }
   )
 
   return {
     prepareTransactionsResult: await prepareTransactions({
       feeCurrencies,
       baseTransactions: rawShortcutTransactionsToTransactionRequests(transactions),
-      isGasSubsidized: isGasSubsidizedForNetwork(token.networkId),
-      origin: `earn-${shortcutId}`,
+      isGasSubsidized: isGasSubsidizedForNetwork(networkId),
+      origin: 'earn-withdraw',
     }),
-    swapTransaction: undefined,
   }
 }
 
-export function usePrepareTransactions(mode: EarnActiveMode) {
-  const prepareTransactions = useAsyncCallback(
-    mode === 'withdraw' ? prepareWithdrawTransactions : prepareDepositTransactions,
-    {
-      onError: (err) => {
-        const error = ensureError(err)
-        Logger.error(TAG, 'usePrepareWithdrawTransactions', error)
-      },
-    }
+export async function prepareClaimTransactions({
+  pool,
+  walletAddress,
+  feeCurrencies,
+  hooksApiUrl,
+  rewardsPositions,
+}: {
+  pool: EarnPosition
+  walletAddress: Address
+  feeCurrencies: TokenBalance[]
+  hooksApiUrl: string
+  rewardsPositions: Position[]
+}) {
+  const { appId, networkId } = pool
+
+  // Prepare the claim transactions if there are rewards positions
+  const claimTransactions = await Promise.all(
+    rewardsPositions.map(async (position): Promise<RawShortcutTransaction[]> => {
+      const { transactions }: { transactions?: RawShortcutTransaction[] } =
+        await triggerShortcutRequest(hooksApiUrl, {
+          address: walletAddress,
+          appId,
+          networkId,
+          shortcutId: 'claim-rewards',
+          ...position.shortcutTriggerArgs?.['claim-rewards'],
+        })
+      return transactions ?? [] // Default to an empty array if rewardsPositions is undefined
```

Comment:
little confused by this comment, if rewardPositions is undefined, this fn would never be executed right?